File: app/src/main/java/com/example/macarus0/walkiewalkie/data/WalkPhotoDao.java
package com.example.macarus0.walkiewalkie.data;

import android.arch.lifecycle.LiveData;
import android.arch.persistence.room.Dao;
import android.arch.persistence.room.Insert;
import android.arch.persistence.room.Query;

import java.util.List;

@Dao
public abstract class WalkPhotoDao {

    @Query("select * from walkphoto where walkId = :walkId")
    public abstract LiveData<List<WalkPhoto>> getAllWalkPhotos(long walkId);

    @Insert
    public abstract void addPhoto(WalkPhoto walkPhoto);

    @Query("delete from walkphoto where walkphoto.photoId = :photoId")
    public abstract void deletePhoto(long photoId);
}

File: app/src/main/java/com/example/macarus0/walkiewalkie/data/WalkPhotoDao.kt
package com.example.macarus0.walkiewalkie.data

import androidx.lifecycle.LiveData
import androidx.room.Dao
import androidx.room.Insert
import androidx.room.Query

@Dao
abstract class WalkPhotoDao {

    @Query("select * from walkphoto where walkId = :walkId")
    abstract fun getAllWalkPhotos(walkId: Long): LiveData<List<WalkPhoto>>

    @Insert
    abstract fun addPhoto(walkPhoto: WalkPhoto)

    @Query("delete from walkphoto where walkphoto.photoId = :photoId")
    abstract fun deletePhoto(photoId: Long)
}
